tests/mocks/core/services/request/RequestMock.php

```diff
@@ -0,0 +1,211 @@
+<?php
+
+namespace EventEspresso\tests\mocks\core\services\request;
+
+use EventEspresso\core\domain\services\contexts\RequestTypeContextCheckerInterface;
+use EventEspresso\core\services\request\Request;
+
+/**
+ * Class RequestMock
+ * Description
+ *
+ * @package EventEspresso\tests\mocks\core\services\request
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class RequestMock extends Request
+{
+
+    /**
+     * $_GET parameters
+     *
+     * @var array $get
+     */
+    public $get;
+
+    /**
+     * $_POST parameters
+     *
+     * @var array $post
+     */
+    public $post;
+
+    /**
+     * $_COOKIE parameters
+     *
+     * @var array $cookie
+     */
+    public $cookie;
+
+    /**
+     * $_SERVER parameters
+     *
+     * @var array $server
+     */
+    public $server;
+
+    /**
+     * $_REQUEST parameters
+     *
+     * @var array $request
+     */
+    public $request;
+
+    /**
+     * @var RequestTypeContextCheckerInterface
+     */
+    public $request_type;
+
+    /**
+     * IP address for request
+     *
+     * @var string $ip_address
+     */
+    public $ip_address;
+
+    /**
+     * @var string $user_agent
+     */
+    public $user_agent;
+
+    /**
+     * true if current user appears to be some kind of bot
+     *
+     * @var bool $is_bot
+     */
+    public $is_bot;
+
+
+    public function __construct(array $get, array $post, array $cookie, array $server, $ip_address = '0.0.0.0')
+    {
+        $this->get = $get;
+        $this->post = $post;
+        $this->cookie = $cookie;
+        $this->server = $server;
+        $this->request = array_merge($this->get, $this->post);
+        $this->ip_address = $this->visitorIp($ip_address);
```

Comment:
This will immediately get overridden in the call to parent::__construct won't it?  Should it be moved below that call?

Reply:
Originally I had no constructor at all for the `RequestMock` class, but the constructor in `Request` failed to set the overridden properties on  `RequestMock` ie; `$this->post = $post;` in `Request::__construct()` set the value for `Request->post` but NOT the value for `RequestMock->post`.

I was surprised by this as you may be now, but it was necessary to setup the `RequestMock` class the way it is to prevent it's properties from all being `null`.